File: problems/01_arrays_and_hashing/0217_contains_duplicate/solution.py
"""
Problem: 217. Contains Duplicate
Difficulty: Easy
URL: https://leetcode.com/problems/contains-duplicate/

【解法演進】暴力 O(n²) → 排序 O(n log n) → 雜湊集合 O(n)（最佳）
【最佳解核心思路】
    維護一個集合 seen，走訪陣列時若 x 已在 seen 內代表重複，直接回傳 True；
    否則加入 seen 繼續。集合查詢均攤 O(1)。
【複雜度】時間：O(n)  空間：O(n)
"""

from typing import List

class Solution:
    # 最佳解：HashSet
    def containsDuplicate(self, nums: List[int]) -> bool:
        pass

    def containsDuplicate_brute(self, nums: List[int]) -> bool:
        pass

    def containsDuplicate_sort(self, nums: List[int]) -> bool:
        pass


if __name__ == "__main__":
    solution = Solution()

    # Test 1
    # result = solution.solve(...)
    # assert result == expected

    print("All tests passed!")

File: problems/01_arrays_and_hashing/0217_contains_duplicate/solution.swift
// Problem: 217. Contains Duplicate
// Difficulty: Easy
// URL: https://leetcode.com/problems/contains-duplicate/
//
// 【解法演進】暴力 O(n²) → 排序 O(n log n) → HashSet O(n)（最佳）
// 【最佳解核心思路】
//   走訪陣列維護 seen 集合，遇到已見元素即回傳 true。
// 【複雜度】時間：O(n)  空間：O(n)

import Foundation

class Solution {
    // 最佳解：HashSet
    func containsDuplicate(_ nums: [Int]) -> Bool {
        // TODO: implement
        return false
    }

    // 暴力解：兩兩比較
    func containsDuplicate_brute(_ nums: [Int]) -> Bool {
        // TODO: implement
        return false
    }

    // 排序後比鄰
    func containsDuplicate_sort(_ nums: [Int]) -> Bool {
        // TODO: implement
        return false
    }
}

func checkBool(_ actual: Bool, _ expected: Bool, _ name: String) {
    if actual == expected { print("✓ \(name)") }
    else { print("✗ \(name): expected \(expected) but got \(actual)") }
}

// MARK: - Tests
// let s = Solution()

// Test 1
// let result = s.solve(...)

print("All tests passed!")
